source/features/hide-inactive-deployments.js

```diff
@@ -0,0 +1,18 @@
+import select from 'select-dom';
+
+export default function () {
+	const deployments = select.all('.discussion-item .deployment-meta');
```

Comment:
Couldn't this be changed to `.discussion-item .deployment-meta.is-inactive` and skip checks inside loop?

Reply:
If the last deployment message is invalid, I don't want to hide it. That's why I have to go through all deployments, and only hide them if they are invalid and not the last deployment.

Reply:
I meant to skip the `isInactive` check (by only looping through inactive `deployment-item`s) not the `isLastDeployment` check!

Reply:
The `isLastDeployment` check refers to all deployments, inactive and active ones.

For example:

```
inactive -> hidden
inactive -> hidden
inactive -> hidden
active
```

```
inactive -> hidden
inactive
```

```
inactive -> hidden
inactive -> hidden
active
active
```

Not the last inactive should be shown, but the last deployment (even if it is inactive).

However, I could save the amount of deployment items and loop through the inactive ones then.

Reply:
Now I understand what you are trying to do. I thought the `.is-inactive` class was in `.deployment-meta`.

Can you try this:
``` js
const deployments = select.all('.discussion-item .deployment-meta');
deployments.pop(); // don't hide the last deployment, even if it is inactive

for (const deployment of deployments) {
	const discussionItem = deployment.closest('.discussion-item');
	const isInactiveDeployment = select.exists('.is-inactive', deployment);
 
	if (isInactiveDeployment) {
		discussionItem.setAttribute('data-rgh-inactive-deployment', 'true');
	}
}
```